app/webpacker/components/RegistrationsV2/Registrations/Competitors.jsx

```diff
@@ -101,164 +150,68 @@ export default function Competitors({
-                <div ref={isUser ? userRowRef : undefined}>
-                  {registration.user.wca_id ? (
-                    <a
-                      href={personUrl(registration.user.wca_id)}
+        <Table.Header>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <Table.Row key={headerGroup.id}>
+              {headerGroup.headers.map((header) => (
+                <Table.HeaderCell
+                  key={header.id}
+                  sorted={expandSortTerm(header.column.getIsSorted())}
+                  onClick={
+                    header.column.getCanSort()
+                      ? header.column.getToggleSortingHandler()
+                      : () => onEventClick(header.column.id)
+                  }
+                  textAlign={header.column.columnDef.meta?.style?.textAlign}
+                >
+                  {flexRender(header.column.columnDef.header, header.getContext())}
+                </Table.HeaderCell>
+              ))}
+            </Table.Row>
+          ))}
+        </Table.Header>
+        <Table.Body>
+          {table.getSortedRowModel().rows.length > 0 ? (
+            table.getSortedRowModel().rows.map((row) => {
+              const isUser = row.original.user_id === userId; 
+              return (
+                <Table.Row active={isUser}>
+                  {row.getVisibleCells().map((cell, index) => (
+                    <Table.Cell
+                      key={cell.id}
+                      textAlign={cell.column.columnDef.meta?.style?.textAlign}
                     >
-                      {registration.user.name}
-                    </a>
-                  ) : (
-                    registration.user.name
-                  )}
-                </div>
-              </Table.Cell>
-              <Table.Cell>
-                <Flag
-                  className={registration.user.country.iso2.toLowerCase()}
-                />
-                {countries.byIso2[registration.user.country.iso2].name}
+                      <div ref={(isUser && (index === 0)) ? userRowRef : undefined}>
+                        {flexRender(cell.column.columnDef.cell, cell.getContext())}
+                      </div>
+                    </Table.Cell>
+                  ))}
+                </Table.Row>
+              );
+            })
+          ) : (
+            <Table.Row>
+              <Table.Cell colSpan={table.getAllColumns().length} textAlign='center'>
+                {I18n.t('competitions.registration_v2.list.empty')}
               </Table.Cell>
-              {eventIds.map((id) => (
-                <Table.Cell
-                  textAlign="center"
-                  key={`registration-table-row-${registration.user.id}-${id}`}
+            </Table.Row>
+          )}
+        </Table.Body>
+        <Table.Footer>
+          {table.getFooterGroups().map(footerGroup => (
+            <Table.Row key={footerGroup.id}>
+              {footerGroup.headers.map(header => (
```

Comment:
Do they really call this `footerGroup.headers`?!

Reply:
Unfortunately. https://tanstack.com/table/v8/docs/framework/react/examples/basic